### PR TITLE
josema/ch93710/fix-wrong-username-error

### PR DESCRIPTION
**Pull request by josemazo**
GH issue [CARTOframes#1466](https://github.com/CartoDB/cartoframes/issues/1466)
CH story [ch93719](https://app.clubhouse.io/cartoteam/story/93710/fix-wrong-username-error)

### CODE REVIEW

**Review by Jesus89 — CHANGES_REQUESTED** [review]
Good approach! However, some things can be improved in the implementation:

- Filter only "not_found" CartoException in the `not_found` decorator. The other CartoExceptions should reach the user
- Move the decorator to the context_manager
- Decorate only the carto-python request methods:
  - https://github.com/CartoDB/cartoframes/blob/develop/cartoframes/io/managers/context_manager.py#L57
  - https://github.com/CartoDB/cartoframes/blob/develop/cartoframes/io/managers/context_manager.py#L60

**Comment by josemazo** [comment]
Re-CR, please!

Yes, the `NotFoundException` stuff is totally necessary, thank you!

File: cartoframes/io/managers/context_manager.py [hunk 1]
```diff
@@ -43,6 +43,21 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 
+def not_found(func):
+    def decorator_func(*args, **kwargs):
+        try:
+            return func(*args, **kwargs)
+
+        except CartoException as e:
```
Comment (Jesus89) [inline cartoframes/io/managers/context_manager.py:51]: I would simplify this to

```py
except NotFoundException:
    raise Exception('Resource not found')
```

Comment (josemazo) [reply cartoframes/io/managers/context_manager.py:51]: That doesn't work :(

The exception is a `CartoException`, created with a `NotFoundException` as the only parameter, so we need to inspect the `args`.

**Review by Jesus89 — APPROVED** [review]

**Pull request merged**